lib/outline.vim

```diff
@@ -87,32 +87,35 @@ enddef
 
 
 def GoToDefinition()
-    # Search item in the Outline side-window first!
-    var curr_line_nr = max([1, line('.') - len(title)])
     var curr_line = getline('.')
-    var counter = len(Outline[0 : curr_line_nr - 1]
-                \ -> filter($"v:val ==# '{curr_line}'"))
-
-    # TODO: check if you can replace wincmd p with some builtin function
-    # Obs! This trigger the BufEnter event!
-    # This means that there will be an immediate (wrong) RefreshWindow()
-    wincmd p
-
-    # The number of jumps needed to reach the target item are counted from the
-    # beginning of the file
-    cursor(1, 1)
-    for ii in range(counter)
-        # TODO This looks for a regular expression not for the literal string!
-        search($'\V{curr_line}', "W")
-    endfor
-    # You moved the cursor, so to be correct you must RefreshWindow() again
-    RefreshWindow()
+    var prev_windid = get(w:, 'PreWinId', 0)
+    if len(curr_line) > 0 && prev_windid != 0
+        if win_gotoid(prev_windid)
+            var i = 1
+            while i <= line('$')
+                if getline(i) == curr_line
+                    break
+                else
+                    i += 1
+                endif
+            endwhile
+            normal! m`
+            exe ':' .. i
+            if i > line('$')
+                echom 'Wrn: [Outline] Content of prev window seems changed'
+            endif
+        else
+            echom 'Err: [Outline] Not found prev window'
+        endif
+    else
+        echom 'Err: [Outline] No or Not a suitable target to process'
+    endif
```

Comment:
I tested this part and it won't work when there are duplicated entries in the outline. Think for example to polymorphism ;-)

Example

class First
def my_function()

class Second
def my_function()

If in the outline you hit enter on the second my_function() it will jump to the first occurrence of my_function() in the calling buffer.

In my implementation I used the counter to overcome this problem.